frontend/src/utils/url.js

```diff
@@ -37,22 +37,26 @@ export function generateUrl(existingSettings, newSettings, path) {
 }
 
 export function qcQueryChanged(before, after) {
-  // Compares ?qcLabel and ?qcRun in the URL to tell when new data will be needed
-  // Invoke while watching vue-router.route.query
+  // Compares ?idProduct in the URL to tell when new data will
+  // be needed. Invoke while watching vue-router.route.query
   if (
-    (after.qcLabel || after.qcRun)
+    (after.idProduct)
```

Comment:
Do we need to test for `after` being defined here (or is it always defined) so that the code that calls this function can drop all tests.

Reply:
This and the else-if below satisfy one condition in the tests, where `before` is undefined and `after` is an empty object. I don't think it matters outside of testing - i.e. we don't need to refresh the QcView if we went from undef to empty.